dev/src/index.ts

```diff
@@ -530,10 +527,11 @@ export class Firestore implements firestore.Firestore {
       if (url.port !== '' && settings.port === undefined) {
         settings.port = Number(url.port);
       }
-      // We need to remove the `host` setting, in case a user calls `settings()`,
-      // which will again enforce that `host` and `servicePath` are not both
-      // specified.
+      // We need to remove the `host` and `apiEndpoint` setting, in case a user
+      // calls `settings()`, which will compare the the provided `host` to the
+      // existing host stored on `servicePath`.
       delete settings.host;
+      delete settings.apiEndpoint;
```

Comment:
I think we can now delete the code that also does this in the constructor.

By the way, I am still a little uneasy about the order of operations here:

```
process.env.FIRESTORE_EMULATOR_HOST = 'chicago';
const firestore = new Firestore({host: 'sanfrancisco'});
// Now talking to 'chicago'

vs:

process.env.FIRESTORE_EMULATOR_HOST = 'chicago';
const firestore = new Firestore()
firstore.setting({host: 'sanfrancisco'});
// Now talking to 'sanfrancisco'
```

Maybe all the constructor code should be moved to `validateAndApplySettings`, where we can then use a unified and sane behavior?

Reply:
moved the deletion code over to `validateAndApplySettings`. However, we need to keep the env var check in the constructor in order to preserve existing behavior where the env var overrides anything passed into the constructor.